airbyte-cdk/bulk/toolkits/extract-jdbc/src/main/kotlin/io/airbyte/cdk/jdbc/JdbcConnectionFactory.kt

```diff
@@ -26,12 +26,15 @@ private val log = KotlinLogging.logger {}
 class JdbcConnectionFactory(
     val config: JdbcSourceConfiguration,
 ) : Supplier<Connection> {
+
+    fun ensureTunnelSession(): TunnelSession =
+        tunnelSessions.computeIfAbsent(config) {
+            val remote = SshdSocketAddress(it.realHost.trim(), it.realPort)
+            createTunnelSession(remote, it.sshTunnel, it.sshConnectionOptions)
+        }
```

Comment:
Lifted this out of `get()` below, because we're going to need this for tunneling the debezium connections.